app/cmd/sync_agent.go

```diff
@@ -87,10 +84,7 @@ func startSyncAgent(c *cli.Context) error {
 		return errors.Wrap(err, "failed to listen")
 	}
 
-	server := grpc.NewServer()
-	ptypes.RegisterSyncAgentServiceServer(server, syncagentrpc.NewSyncAgentServer(start, end, replicaAddress,
-		volumeName, replicaInstanceName))
-	reflection.Register(server)
```

Comment:
NIT: Is this necessary? All other server launching are still using this old pattern

Reply:
I struggled a bit here, as each server seems to me to be launched a bit differently. I ultimately decided to move this logic to `pkg/snyc/rpc/server.go/` because the others make the `grpc.NewServer` call and add the interceptor within their respective packages.